### PR TITLE
Source/sha: Fix signed-integer-overflow [WIP]

### PR DESCRIPTION
**Pull request by glebm**
DO NOT MERGE

This seems to break existing saves (checksum doesn't match) and I'm not yet sure why.

### CODE REVIEW

File: Source/sha.cpp [hunk 1]
```diff
@@ -50,9 +50,9 @@ void SHA1Input(SHA1Context *context, const char *message_array, int len)
 
 void SHA1ProcessMessageBlock(SHA1Context *context)
 {
-	int i, temp;
-	int W[80];
-	int A, B, C, D, E;
+	DWORD i, temp;
+	DWORD W[80];
+	DWORD A, B, C, D, E; // Only works with `int`, perhaps it relies on sign-extension???
```
Comment (glebm) [inline Source/sha.cpp]: @AJenbo Are our save files compatible with original Diablo?
If so, then the SHA1 implementation in OG Diablo is wrong :sweat_smile:

Comment (qndel) [reply Source/sha.cpp]: look at asm? :p

**Comment by AJenbo** [comment]
Yep blizzard has a broken SHA1, maybe we should document this fact better.

**Comment by glebm** [comment]
Fixed in DevilutionX but doubt it'll be bin-exact here
https://github.com/diasurgical/devilutionX/pull/343

**Pull request closed without merging**